sdk/core/core-amqp/src/requestResponseLink.ts

```diff
@@ -98,65 +85,38 @@ export class RequestResponseLink implements ReqResLink {
     if (!options) options = {};
 
     if (!options.timeoutInSeconds) {
-      options.timeoutInSeconds = 10;
+      options.timeoutInSeconds = Constants.defaultOperationTimeoutInSeconds;
     }
 
-    let count: number = 0;
     const aborter: AbortSignalLike | undefined = options && options.abortSignal;
 
     const sendRequestPromise = () =>
       new Promise<AmqpMessage>((resolve: any, reject: any) => {
-        let waitTimer: any;
         let timeOver: boolean = false;
         type NormalizedInfo = {
           statusCode: number;
           statusDescription: string;
           errorCondition: string;
         };
 
-        count++;
-        if (count !== 1) {
-          // Generate a new message_id every time after the first attempt
-          request.message_id = generate_uuid();
-        } else if (!request.message_id) {
-          // Set the message_id in the first attempt only if it is not set
-          request.message_id = generate_uuid();
-        }
-
-        const rejectOnAbort = () => {
-          const address = this.receiver.address || "address";
-          const requestName = options!.requestName;
-          const desc: string =
-            `[${this.connection.id}] The request "${requestName}" ` +
-            `to "${address}" has been cancelled by the user.`;
-          log.error(desc);
-          const error = new AbortError(
-            `The ${requestName ? requestName + " " : ""}operation has been cancelled by the user.`
-          );
-
-          reject(error);
-        };
-
-        const onAbort = () => {
-          // remove the event listener as this will be registered next time someone makes a request.
+        const actionAfterTimeout = () => {
+          timeOver = true;
           this.receiver.removeListener(ReceiverEvents.message, messageCallback);
-          // safe to clear the timeout if it hasn't already occurred.
-          if (!timeOver) {
-            clearTimeout(waitTimer);
+          if (aborter) {
+            aborter.removeEventListener("abort", onAbort);
           }
-          aborter!.removeEventListener("abort", onAbort);
-
-          rejectOnAbort();
```

Comment:
In order to help the review process, can you please consider keeping the previous order of the various callbacks/helper functions here? This would reduce the the number lines in the diff.

Reply:
Also, due to the re-arrangement of the code here, we are introducing bugs.
For example, in the new order, the timer is being started first before the check on whether the abort signal was fired.
If the abort signal was fired `rejectOnAbort()` gets called and the timer never gets cleared.

In the previous order since the abort signal was checked first, we didnt have to worry about clearing the timer

Reply:
Actually rejectOnAbort() happens asynchronously right? Shouldn't it anyway be clearing the timer?

Reply:
Timer is actually cleared as part of `onAborted` callback already, looks like the bug concern doesn't hold true. 
So don't think there was any issue, but reverted code arrangement anyway for easier review.

Reply:
Please re-read the comment.

The concern was not for when the abort signal was fired in the middle of the async process of init() or the actual send().

The concern was for when the abort signal was already fired by the time we enter the code path here.

When abort signal is fired in the middle of a async process, `onAborted` gets called and everything is handled well.

If the abort signal was fired before the code path is entered here, then `onAborted` will never get called, because the signal would have been fired before we could attach `onAborted` as the event listener. This is why we have the explicit checks of `if (abortSignal && abortSignal.aborted)`

Reply:
> This is why we have the explicit checks of if (aborter.aborted)

Oh okay, I thought if an abort signal being fired always results in the `abort` event getting fired and it getting handled by the `onAbort` handler. It wasn't clear that "abort signal firing" meant the input parameters would already have the value set and no events are fired.
Thanks for clarifying! If this is indeed the case, this makes sense and so does the need for synchronized checks.

Reply:
Also the inconsistency in usage is leading to problems, slow downs as well, especially for invocations/usage of abort signal and event handling since it's a new thing. 
It maybe worth considering having an issue to streamline the code organization where possible, like as part of https://github.com/Azure/azure-sdk-for-js/pull/4319#discussion_r304173981 but factoring in abort too.

Reply:
> Also the inconsistency in usage across operations is leading to problems, slow downs as well, especially for invocations/usage of abort signal and event handling since it's a new thing

The invocations and usage of abort signal has been consistent across Event Hubs and core-amqp in the master branch. @ShivangiReja and @chradek have made sure of that. We have also been consistent regarding removing event listeners and clearing timers. 
- We always do an early exit if the `aborted` property on the signal is `true`
- We always add an event listener to the `abort` event on the abort signal
- We always remove all event listeners and clear timer when returning regardless of whether the promise is resolved or rejected.

As long as the above is taken care of, we are good and consistent enough. 

> It maybe worth considering having an issue to streamline the code organization where possible, like as part of #4319 (comment) but factoring in abort too.

The order of creating callbacks, attaching listeners, starting timers, clearing listeners & timers, abort signal: all of this works as expected in the master branch. I don't see the need to stream line them any further.

Aside: https://github.com/Azure/azure-sdk-for-js/pull/4319#discussion_r304173981 was a guess at why @chradek's prior suggestion might not work and not a suggestion to re-order anything

Reply:
I see uses/naming such as below that need fixing at least

```
const aborter: AbortSignalLike | undefined = options && options.abortSignal;
```

```
const abortSignal: AbortSignalLike | undefined = options.abortSignal;
```